### PR TITLE
vrp: fix local build flow, add to examples CI validation.

### PR DESCRIPTION
**Pull request by htuch**
Changes to .dockerignore broke the flow described at
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/google_vrp#rebuilding-the-docker-image.

This PR fixes and adds a CI verify-example test.

Risk level: Low
Testing: New verify-examples case.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+!/build*
```
Comment (phlax) [inline .dockerignore:2]: im wondering if we can be any more specific - iirc this dir can get large and then be sent to the docker daemon - might not matter so much

Comment (htuch) [reply .dockerignore:2]: Would `build_envoy` be OK?

Comment (phlax) [reply .dockerignore:2]: i dont rem exactly - my recall is that the `build` folder is used - but perhaps im confusing it (with inside the `build` container) 

If its not used already then just using `build` would be fine.

 If it is - then, yep, using a different folder or pointing to subdir would be best

**Review by phlax — APPROVED** [review]
lgtm

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**